### PR TITLE
YUM install

### PR DESCRIPTION
**Pull request by hakonhall**

### CODE REVIEW

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/process/ProcessImpl.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+package com.yahoo.vespa.hosted.node.admin.task.util.process;
+
+import com.yahoo.vespa.hosted.node.admin.task.util.file.IOExceptionUtil;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class ProcessImpl implements ProcessApi {
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/process/ProcessImpl.java]: This seems to be a subset of `Command`?

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/yum/Yum.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            }
+        }
+
+        public Install enableRepo(String repo) {
+            enabledRepo = Optional.of(repo);
+            return this;
+        }
+
+        public boolean converge() {
+            if (packages.stream().allMatch(Yum.this::isInstalled)) {
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/yum/Yum.java]: Would be nice if we could execute the command once to get a list of all installed packages and then just compare two `Set`s.

File: node-admin/src/test/java/com/yahoo/vespa/hosted/node/admin/task/util/yum/YumTest.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    @Test
+    public void testInstall() {
+        TaskContext taskContext = mock(TaskContext.class);
+        TestCommandSupplier commandSupplier = new TestCommandSupplier(taskContext);
+
+        commandSupplier.expectCommand("yum list installed package-1", 0, "");
+        commandSupplier.expectCommand("yum list installed package-2", 1, "");
+        commandSupplier.expectCommand(
+                "yum install --assumeyes --enablerepo=repo-name package-1 package-2",
```
Comment (freva) [inline node-admin/src/test/java/com/yahoo/vespa/hosted/node/admin/task/util/yum/YumTest.java]: Do we want to install both if we already know one of them is installed?

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/process/ProcessApi.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.node.admin.task.util.process;// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/process/ProcessApi.java]: Duplicate copyright header.

Comment (hakonhall) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/process/ProcessApi.java]: Fixed

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/process/ProcessImpl.java (hunk 1 above)
Comment (hakonhall) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/process/ProcessImpl.java]: It's a left-over from an earlier version, let me remove these.

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/yum/Yum.java (hunk 2 above)
Comment (hakonhall) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/yum/Yum.java]: Yes, but that would require parsing the output. Now, it's just an exit code.

File: node-admin/src/test/java/com/yahoo/vespa/hosted/node/admin/task/util/yum/YumTest.java (hunk 3 above)
Comment (hakonhall) [reply node-admin/src/test/java/com/yahoo/vespa/hosted/node/admin/task/util/yum/YumTest.java]: Could we risk upgrading the package left out?

**Review by freva — APPROVED** [review]

**Pull request merged**